packages/config-plugins/src/ios/SplashScreen.ts

```diff
@@ -38,8 +38,11 @@ export function getSplashScreen(config: ExpoConfig): IosSplashScreenConfig | und
 }
 
 export async function setSplashScreenAsync(config: ExpoConfig, projectRoot: string) {
+  const majorVersionString = config.sdkVersion?.split('.').shift();
   const splashScreenIsSupported =
-    config.sdkVersion === '39.0.0' || config.sdkVersion === '40.0.0' || !config.sdkVersion;
+    (majorVersionString && Number(majorVersionString) >= 39) ||
+    /** UNVERSIONED CASE */ !majorVersionString;
+
```

Comment:
Should also improve the warning message so users are aware that the plugin is disabled on the current expo version, and maybe also inform them to upgrade.